libcontainer/cgroups/systemd/unified_hierarchy.go

```diff
@@ -167,18 +171,42 @@ func (m *UnifiedManager) Destroy() error {
 	if m.Cgroups.Paths != nil {
 		return nil
 	}
+	dirPath, err := m.GetUnifiedPath()
+	if err != nil {
+		return err
+	}
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
 	dbusConnection, err := getDbusConnection()
 	if err != nil {
 		return err
 	}
-	dbusConnection.StopUnit(getUnitName(m.Cgroups), "replace", nil)
-	if err := cgroups.RemovePaths(m.Paths); err != nil {
-		return err
+
+	unitName := getUnitName(m.Cgroups)
+	statusChan := make(chan string, 1)
+
+	if _, err := dbusConnection.StopUnit(unitName, "replace", statusChan); err == nil {
+		select {
+		case s := <-statusChan:
+			if s != "done" {
+				logrus.Warnf("error stopping systemd unit `%s`: got `%s`. Continuing...", unitName, s)
+			}
+		case <-time.After(time.Second):
+			logrus.Warnf("Timed out while waiting for StopUnit(%s) completion signal from dbus. Continuing...", unitName)
+		}
+	} else {
+		// TODO: When we destroy cgroup, the initProcess has been killed, so we always get `Unit **** not loaded` err.
+		// Because the unit is created by StartTransientUnit method,
+		// so when process exited, the cgroup path may has been deleted by systemd.
+		// Because of this, we should not return the error.
+		// But if the system is very busy, we may not get this error by chance.
+		// Shall we need to always warn this information?
+		logrus.Warnf("error stopping systemd unit `%s`: got `%s`. Continuing...", unitName, err)
```

Comment:
@kolyshkin 
Shall we need to always warn this information?
When we destroy cgroup, the initProcess has been killed, so we will always get `Unit **** not loaded` err.
Because the unit is created by StartTransientUnit method,
so when process exited, the cgroup path may has been deleted by systemd.
Because of this, we should not return the error.
But if the system is very busy, we may not get this error by chance.

Reply:
I think I should move this commit to #2329 , and discuss in there.
WDYT @kolyshkin